core/tests/puppeteer-acceptance-tests/utilities/user/super-admin.ts

```diff
@@ -99,6 +176,203 @@ export class SuperAdmin extends BaseUser {
     showMessage(`User ${username} does not have the ${role} role!`);
     await this.goto(currentPageUrl);
   }
+
+  /**
+   * Unassigns a role from a user.
+   * @param {string} username - The username of the user.
+   */
+  async unassignRoleFromUser(username: string, role: string): Promise<void> {
+    role = role.replace(/ /g, '-');
+    await this.goto(AdminPageRolesTab);
+    await this.page.waitForSelector(roleEditorInputField);
+    await this.type(roleEditorInputField, username);
+    await this.clickOn(roleEditorButtonSelector);
+    await this.page.waitForSelector('div.justify-content-between');
+    await this.page.waitForSelector(
+      `.e2e-test-${role}-remove-button-container`
+    );
+
+    const deleteRoleButton = await this.page.$(
+      `.e2e-test-${role}-remove-button-container`
+    );
+    if (!deleteRoleButton) {
+      throw new Error(`User does not have the ${role} role!`);
+    }
+
+    await this.waitForElementToBeClickable(deleteRoleButton);
+    await deleteRoleButton.click();
+    showMessage(`Role ${role} has been removed from user ${username}`);
+    return;
+  }
+
+  /**
+   * Selects a role.
+   * @param {string} role - The role to select.
+   */
+  async selectRole(role: string): Promise<void> {
+    await this.navigateToAdminPageRolesTab();
+    role = role.replace(/\b\w/g, char => char.toUpperCase());
+    await this.clickOn(role);
+  }
+
+  /**
+   * This function checks if the allocated actions for a role are present on the page.
+   * @param {string[]} actions - The actions to check for.
+   */
+  async expectRoleToHaveAllocatedActions(actions: string[]): Promise<void> {
+    for (const action of actions) {
+      const isActionPresent = await this.isTextPresentOnPage(action);
+      if (!isActionPresent) {
+        throw new Error(`Action ${action} is not allocated to the role`);
+      }
+    }
+    showMessage(`${actions} is/are allocated to the role`);
+  }
+
+  /**
+   * Checks if the specified users are assigned to the current role.
+   * @param {string[]} users - An array of usernames to check.
+   */
+  async expectRoleToHaveAssignedUsers(users: string[]): Promise<void> {
+    await this.clickOn(' Assigned users ');
+    await this.page.waitForTimeout(2000);
```

Comment:
Is waitForTimeout the only way? Can you use waitForNavigation instead? We try to avoid using waitForTimeout.

Reply:
removed.